File: 2020_advent/Day_5-[Binary_Boarding]/php/exercise_1.php
<?php

$input = file_get_contents("../input_data.txt");

$arr = explode("\r\n", $input);
$boards = [];

foreach($arr as $board_seat) {
    $board_data = str_replace(['F', 'B', 'L', 'R'], [0, 1, 0, 1], $board_seat);
    $board_data = str_split($board_data, 7);
    $board_seat_id = bindec($board_data[0]) * 8 + bindec($board_data[1]);
    $boards[] = $board_seat_id;
}
echo max($boards);

//Result 801

File: 2020_advent/Day_5-[Binary_Boarding]/python/exercise_1.py
input = open('input_data.txt', 'r')

def str_arr_replace(phrase, char_to_replace):
    for key, value in char_to_replace.items():
        phrase = phrase.replace(key, value)
    return phrase

boards = []
for seat in input:
    board_data = str_arr_replace(seat, {'F':'0', 'B':'1', 'L':'0', 'R':'1'})
    board_row = int(board_data[:7], 2)
    board_col = int(board_data[7:10], 2)
    board_seat_id = board_row * 8 + board_col
    boards.append(board_seat_id)

print(max(boards))

# Result 801